crates/bevy_ui/src/ui_node.rs

```diff
@@ -391,3 +391,48 @@ pub struct CalculatedClip {
     /// The rect of the clip
     pub clip: bevy_sprite::Rect,
 }
+
+/// The border radius of the node
+/// 
+/// This doesn't require a [`Border`] component
+#[derive(Component, Default, Copy, Clone, Debug, Reflect)]
+#[reflect(Component)]
+pub struct BorderRadius {
+    pub top_left: f32,
+    pub bottom_left: f32,
+    pub top_right: f32,
+    pub bottom_right: f32,
+}
+
+impl BorderRadius {
+    pub fn all(border_radius: f32) -> Self {
+        Self {
+            top_left: border_radius,
+            bottom_left: border_radius,
+            top_right: border_radius,
+            bottom_right: border_radius,
+        }
+    }
+
+    pub fn to_array(&self) -> [f32; 4] {
+        [
+            self.top_left,
+            self.bottom_left,
+            self.top_right,
+            self.bottom_right,
+        ]
+    }
+}
+
+/// The visual properties of the node's border
+#[derive(Component, Default, Copy, Clone, Debug, Reflect)]
+#[reflect(Component)]
+pub struct Border {
+    /// The width of the border
+    ///
+    /// This is different from [`Style`] border and it will not cause any displacement inside the node.
+    pub width: f32,
```

Comment:
This (and the border radius) should be a `Val` if possible to enable users to specify sizes on a percent-basis.

Reply:
Since this is one border width for the entire rect at the moment, what do you think a percent value should represent?

Reply:
Hmm. I was thinking "a fraction of the scale of the rectangle", but you're right: this could either be width or height.

Reply:
I'd like to add this, but I haven't implemented it yet because it's unclear what % should mean for border width. I think if this supports having different borders for each side at some point, then it would make more sense to add it.